presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -69,6 +69,19 @@ private Scope(
         }
     }
 
+    public Optional<Scope> getOuterQueryParent()
```

Comment:
This is for `StatementAnalyzer.analyze` assertion. Returned scope should have `outerQuery` somewhere as an ancestor.